File: PY/2434.py
# BEE 2434 - Saldo do Vovo - Python

def balanco(saldo, movimentacoes):
    menor = saldo
    for movimentacao in movimentacoes:
        saldo += movimentacao
        if saldo < menor:
            menor = saldo
    return menor


n_movimentacoes, saldo = input().split()
saldo = int(saldo)
movimentacoes = [int(input()) for i in range(int(n_movimentacoes))]

print((balanco(saldo, movimentacoes)))

File: C/2434.c
#include <stdio.h>

int saldo(int n, int s)
{
    int movimentacao, menor = s;
    for (int i = 0; i < n; i++)
    {
        scanf("%d", &movimentacao);
        s += movimentacao;
        if (s < menor)
        {
            menor = s;
        }
    }
    return menor;
}

int main()
{
    int n, s;
    scanf("%d %d", &n, &s);

    printf("%d\n", saldo(n, s));
    return 0;
}